android/app/src/main/java/org/fossasia/openevent/core/schedule/DayScheduleFragment.java

```diff
@@ -135,36 +128,20 @@ public void onChanged() {
     }
 
     public void filter(List<String> selectedTracks) {
-        if(dayScheduleAdapter == null)
+        if (dayScheduleAdapter == null)
             return;
 
-        realmRepo.getSessionsByDate(date, SortOrder.sortTypeSchedule())
-                .addChangeListener((sortedSessions, orderedCollectionChangeSet) -> {
-                    sessions.clear();
-                    sessions.addAll(sortedSessions);
-
-                    filteredSessions.clear();
-                    if (selectedTracks.isEmpty()) {
-                        filteredSessions.addAll(sessions);
-                    } else {
-                        for(int i = 0; i < sessions.size() ; i++) {
-                            String trackName = sessions.get(i).getTrack().getName();
-                            if (selectedTracks.contains(trackName)) {
-                                filteredSessions.add(sessions.get(i));
-                            }
-                        }
-
-                        if(searchText!=null)
-                            dayScheduleAdapter.filter(searchText);
-                    }
-
-                    if(SortOrder.sortOrderSchedule() == SortOrder.SORT_ORDER_DESCENDING)
-                        Collections.reverse(filteredSessions);
-
-                    dayScheduleAdapter.notifyDataSetChanged();
-
-                    handleVisibility();
-                });
+        dayScheduleFragmentViewModel.getSessionsByTracks(date, selectedTracks).observe(DayScheduleFragment.this, trackSessions -> {
+            filteredSessions.clear();
+            filteredSessions.addAll(trackSessions);
+            if (searchText != null)
+                dayScheduleAdapter.filter(searchText);
```

Comment:
Should this filtering be placed in viewmodel or fragment?

Reply:
Yes, please. The aim is to make the view as dumb as possible and hence all the logic to ViewModel if possible.

Reply:
Ok thanks for the help :)!